src/Core/Scheduler/Providers/TileProvider.js

```diff
@@ -49,47 +41,80 @@ TileProvider.prototype.preprocessDataLayer = function preprocessLayer(layer, vie
     });
 };
 
+const worldQuaternion = new THREE.Quaternion();
 TileProvider.prototype.executeCommand = function executeCommand(command) {
-    var extent = command.extent;
+    const extent = command.extent;
     if (command.requester &&
         !command.requester.material) {
         // request has been deleted
         return Promise.reject(new CancelledCommandException(command));
     }
-
-    var parent = command.requester;
-
+    const layer = command.layer;
+    const builder = layer.builder;
+    const parent = command.requester;
+    const level = (command.level === undefined) ? (parent.level + 1) : command.level;
+
+    const { sharableExtent, quaternion, position } = builder.computeSharableExtent(extent);
+    const south = sharableExtent.south().toFixed(6);
+    const segment = layer.segments || 16;
+    const key = `${builder.type}_${layer.disableSkirt ? 0 : 1}_${segment}_${level}_${south}`;
+
+    let geometry = this.cacheGeometry.get(key);
+    // build geometry if doesn't exist
+    if (!geometry) {
+        const paramsGeometry = {
+            extent: sharableExtent,
+            level,
+            segment,
+            disableSkirt: layer.disableSkirt,
+        };
+
+        geometry = new TileGeometry(paramsGeometry, builder);
+        geometry._count = 0;
+        this.cacheGeometry.set(key, geometry);
```

Comment:
You could do this:
```js
geometry.dispose = () => {
    geometry._count--;
    if (geometry._count == 0) {
       THREE.BufferGeometry.prototype.dispose.call(geometry);
       this.cacheGeometry.delete(key);
    }
}
```
And remove the `count` property. This way the ref-counting feature would be internal to TileProvider.